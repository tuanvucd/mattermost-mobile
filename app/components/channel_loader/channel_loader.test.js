// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import ChannelLoader from './channel_loader';

jest.mock('rn-placeholder', () => ({
    ImageContent: () => {},
}));

describe('ChannelLoader', () => {
    const baseProps = {
        channelIsLoading: true,
        theme: Preferences.THEMES.default,
        actions: {
            handleSelectChannel: jest.fn(),
            setChannelLoading: jest.fn(),
        },
        isLandscape: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<ChannelLoader {...baseProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});