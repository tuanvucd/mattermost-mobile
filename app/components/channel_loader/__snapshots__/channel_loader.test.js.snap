// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelLoader should match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
      null,
      Object {
        "backgroundColor": "#ffffff",
      },
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "paddingLeft": 12,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "opacity": 0.6,
        },
      ]
    }
  >
    <ImageContent
      animate="fade"
      color="rgba(61,60,64,0.15)"
      firstLineWidth="80%"
      hasRadius={true}
      lineNumber={3}
      lineSpacing={5}
      size={32}
      textSize={14}
    />
  </AnimatedComponent>
</View>
`;
